### DISCUSSION TITLE
Questions on the Orbit restructuring

### DISCUSSION THREAD

**Discussion by wilbert-alberts** [eclipse-orbit/orbit-simrel#38] [Q&A]
Hi,
I have been using the Orbit update sites before in order versions of eclipse in order to install fasterxml/json, apache CLI/logging/CSV etc. in my eclipse installation. With these eclipse versions (at least up to 2022-03) I could more or less cherry pick what I needed.

I recently tried the same with eclipse 2024-06. I came to this site: https://download.eclipse.org/tools/orbit/simrel/orbit-aggregation/ That resulted in a couple of questions:

1) I seem to remember that older update sites contained the plugins themselves. Now it seems that for a number of features, the actual plugins are retrieved from a maven update site. Or am I wrong?

2) I used to be able to cherry pick individual features, now (at least for plugins/features like Apache Commons CSV) I need to install Orbit Legacy and Orbit Maven OSGi All. Is there a way to install 'the smaller things' like the log4j, CSV, fasterxml/json alike?

3) Installing Orbit Legagy and Orbit Maven OSVi All in Eclipse 2024-06 seems to brake the graphical ecore (class diagram alike) editor. Did I install Orbit the wrong way or should I raise an issue?

Regards,
Wilbert

**Comment by merks** [comment]
>     1. I seem to remember that older update sites contained the plugins themselves. Now it seems that for a number of features, the actual plugins are retrieved from a maven update site. Or am I wrong?
> 
The features are just a convenience for building the update site.  They are not intended to be used anywhere for any reason.  Only the bundles are intended to be reused elsewhere.  And yes, many bundles are pulled directly from Maven without any changes.  In particular, these ones:

https://download.eclipse.org/tools/orbit/simrel/maven-osgi/

>     2. I used to be able to cherry pick individual features, now (at least for plugins/features like Apache Commons CSV) I need to install Orbit Legacy and Orbit Maven OSGi All. Is there a way to install 'the smaller things' like the log4j, CSV, fasterxml/json alike?
> 

I'm not sure I understand the question.  Generally these Orbit repositories are not intended to be used to directly install any of the bundles or features.  Instead they are intended to be used in your target platform definition to resolve your dependencies.

>     3. Installing Orbit Legagy and Orbit Maven OSVi All in Eclipse 2024-06 seems to brake the graphical ecore (class diagram alike) editor. Did I install Orbit the wrong way or should I raise an issue?

This is not a sensible activity.  None of these bundles provide end-user feature such that you'd want to install them directly into an IDE.  They're libraries used by other plugins for some specific purpose.

**Reply by wilbert-alberts** [reply]
Thanks for the reply. Indeed it seems I have been using the wrong tool here. Yesterday I found out that a target platform can also contain maven dependencies (as long as the m2e functionality has been installed in eclipse). That allows me to specifically select the bundles I need. I will try it that way (especially the fact whether I can use them to develop new stuff in eclipse and to get the stuff build by mvn).

**Reply by merks** [reply]
Keep in mind that often artifacts at maven central are not in the form of OSGi bundles in which case they need to be wrapped and then you can control how they are wrapped with BND instructions:

https://github.com/eclipse-orbit/orbit-simrel/blob/730eaaf1ee2caa6ec921416c9d9341ef66b00778/maven-bnd/tp/MavenBND.target#L69-L80

Properly wrapping isn't not always entirely trivial...
